Makefile

```diff
@@ -82,12 +82,23 @@ server-env: ca-cert-env gen-db-env
 	sleep 2s
 	@$(BLR)
 
-# Start caramel_autosign in the background, save PID to AUTOSIGN_PID
+# Start caramel using pserve in the background, save PID to SERVER_PID
+.PHONY: server.in%
+server-in%: ca-cert-in% gen-db-in%
+	@ $(BOLD); echo "Start new caramel server in the background, sleep 2s to \
+	give it time to start";\
+	$(BLR)
+	setsid $(VENV)/bin/pserve $(CARAMEL_COMMAND_LINE) >/dev/null 2>&1 < /dev/null & \
+	echo $$! > $(SERVER_PID)
+	sleep 2s
+	@$(BLR)
+
+# Start caramel_autosign in the background, save PID to ENV_AUTOSIGN_PID
 .PHONY: autosign%
 autosign%: $(CARAMEL_TOOL) ca-cert% gen-db%
 	@$(BOLD);echo "Start new caramel_autosign in the background";\
 	$(BLR)
-	setsid $(VENV)/bin/caramel_autosign >/dev/null 2>&1 < /dev/null &\
+	setsid $(VENV)/bin/caramel_autosign $(CARAMEL_COMMAND_LINE) >/dev/null 2>&1 < /dev/null &\
 	echo $$! > $(AUTOSIGN_PID)
```

Comment:
I just noticed that multiple parallel make calls will use the same PID here,  so you'll lose some processes.

(AUTOSIGN_PID is per Venv)   

Not sure if it really really matters, but it's good to be aware.

I'd probably make the pid the target of the rule, i e.  "autosign-%.pid"  as a non-phony target, as that would make the ordering "right" and could also make it destroyable afterwards.

That's not to be fixed now, but more an FYI as the current might go weird if you do "make -j32 systest"

Reply:
Actually, looking at the full diff, this is a negative change  as  no concurrency options work with the current setup :/

Reply:
fixed